pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -150,6 +152,7 @@ func readRuntimeObjectAsPipeline(ctx context.Context, namespace string, obj runt
 		// TODO(#6592): Decouple API versioning from feature versioning
 		dryRunObj := obj.DeepCopy()
 		dryRunObj.Name = uuid.NewString() // Use a randomized name for the Pipeline in case there is already another Pipeline of the same name
+		dryRunObj.Namespace = ""          // Make sure the namespace is empty as it won't be used later on
```

Comment:
Yeah my bad, I was too quick on the fix here 😬

Reply:
np! I'm wondering if there are situations where it would make sense to use the namespace of the resolved pipeline rather than the namespace of the pipelinerun? For the cluster resolver, we know it was a valid pipeline in the namespace where it existed; is it possible that it's an invalid pipeline in the namespace of the pipelinerun, and if so, should we run it or not? The situations where I could imagine this happening is if we implement TEP-0085 (per-namespace controller config) (or possibly TEP-0129). I was trying to think of any k8s policies that could cause a pipeline to be valid in one ns but not another, but I'm not sure of any after looking at the k8s docs on policies.